components/eventHandler.ts

```diff
@@ -989,10 +991,10 @@ function saveEvents(
                         (<MurderedBodySeenC2SEvent>event).Value.Witness,
                     )
 
-                    if (event.Timestamp === session.lastKill.timestamp) {
-                        session.killsNoticedBy.add(
-                            (<MurderedBodySeenC2SEvent>event).Value.Witness,
-                        )
+                    if (
+                        !(<MurderedBodySeenC2SEvent>event).Value.IsWitnessTarget
```

Comment:
I'm confused. What does this property actually check for?

Based on it's name, I could infer a few possible meanings:
- Is the witness a target?
- Has the target (kill) been witnessed

The only reason I ask is that in H2/3, you can commit an illegal action in front of a target, they spot you and become a witness, but then you kill them, and your SA comes back (as long as they didn't "tell anyone" - i.e. the orange doesn't spread).
This behavior is not the same in 2016, where if you are spotted, even if it's by your target, it's RIP SA.

This ties back to my original question, it's not clear what this property does. Can we add a docstring on the typedef with a brief explanation?

Reply:
The property is true if the witness is a target.
That means that with this PR, if any body is found by a non-target, you will lose SA and cannot regain it by killing that witness.
I'll add a docstring.

The 2016 logic should be unaffected by this change.
I think it is a bit clearer if you look at the changes per commit.